src/app/vm/vm-creation/template/vm-template-dialog.component.ts

```diff
@@ -60,10 +66,33 @@ export class VmTemplateDialogComponent extends TemplateFilterListComponent imple
   }
 
   public onOk(): void {
-    this.dialogRef.close(this.selectedTemplate);
+    const dialogData = { template: this.selectedTemplate, agreement: false };
+
+    this.templateTagService.getAgreement(this.selectedTemplate ? this.selectedTemplate : this.preselectedTemplate)
+      .subscribe(res => {
```

Comment:
please get rid of multiple subscriptions:
Let's try to use this.dialogRef.close(dialogData)  only once in the code to make the flow cleaner